lib/driver.rb

```diff
@@ -0,0 +1,56 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: [])
+      super(id)
+
+      @name = name
+      @vin = vin
+      raise ArgumentError.new("Invalid VIN") if (vin.length) != 17
+
+      @status = status
+      raise ArgumentError.new("Invalid status") if ![:AVAILABLE, :UNAVAILABLE].include?(@status)
+
+      @trips = trips
+
+    end
+
+    def accept_new_trip(trip)
+      add_trip(trip)
+      @status = :UNAVAILABLE
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating
+      ratings_array = @trips.reject { |trip| trip.rating == nil }.map(&:rating)
+      return 0 if (@trips) == [] || @trips == nil
+      return (ratings_array.sum/ratings_array.length).to_f
+    end
```

Comment:
You should move this check up.  If `@trips` is `nil` you will get an error when you call `@trips.reject`:

```suggestion
    def average_rating
      return 0 if (@trips) == [] || @trips == nil
      ratings_array = @trips.reject { |trip| trip.rating == nil }.map(&:rating)
      return (ratings_array.sum/ratings_array.length).to_f
    end
```